pkg/detectors/gitlab/v1/gitlab.go

```diff
@@ -88,74 +112,51 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 	return results, nil
 }
 
-func (s Scanner) verifyGitlab(ctx context.Context, resMatch string) (bool, map[string]string, map[string]string, error) {
+func VerifyGitlab(ctx context.Context, client *http.Client, baseEndpoint, resMatch string) (bool, map[string]string, error) {
 	// there are 4 read 'scopes' for a gitlab token: api, read_user, read_repo, and read_registry
 	// they all grant access to different parts of the API. I couldn't find an endpoint that every
 	// one of these scopes has access to, so we just check an example endpoint for each scope. If any
 	// of them contain data, we know we have a valid key, but if they all fail, we don't
 
-	client := s.client
-	if client == nil {
-		client = defaultClient
+	req, err := http.NewRequestWithContext(ctx, "GET", baseEndpoint+"/api/v4/user", nil)
+	if err != nil {
+		return false, nil, err
 	}
-	for _, baseURL := range s.Endpoints() {
-		// test `read_user` scope
-		req, err := http.NewRequestWithContext(ctx, "GET", baseURL+"/api/v4/user", nil)
-		if err != nil {
-			continue
-		}
-
-		req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", resMatch))
-		res, err := client.Do(req)
-		if err != nil {
-			return false, nil, nil, err
-		}
-
-		defer res.Body.Close()
 
-		bodyBytes, err := io.ReadAll(res.Body)
-		if err != nil {
-			return false, nil, nil, err
-		}
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", resMatch))
+	res, err := client.Do(req)
+	if err != nil {
+		return false, nil, err
+	}
 
-		analysisInfo := map[string]string{
-			"key":  resMatch,
-			"host": baseURL,
-		}
+	defer res.Body.Close()
 
-		// 200 means good key and has `read_user` scope
-		// 403 means good key but not the right scope
-		// 401 is bad key
-		switch res.StatusCode {
-		case http.StatusOK:
-			return json.Valid(bodyBytes), nil, analysisInfo, nil
-		case http.StatusForbidden:
-			// check if the user account is blocked or not
-			stringBody := string(bodyBytes)
-			if strings.Contains(stringBody, BlockedUserMessage) {
-				return true, map[string]string{
-					"blocked": "True",
-				}, analysisInfo, nil
-			}
+	bodyBytes, err := io.ReadAll(res.Body)
+	if err != nil {
+		return false, nil, err
+	}
 
-			// Good key but not the right scope
-			return true, nil, analysisInfo, nil
-		case http.StatusUnauthorized:
-			// Nothing to do; zero values are the ones we want
-			return false, nil, nil, nil
-		default:
-			return false, nil, nil, fmt.Errorf("unexpected HTTP response status %d", res.StatusCode)
+	// 200 means good key and has `read_user` scope
+	// 403 means good key but not the right scope
+	// 401 is bad key
+	switch res.StatusCode {
+	case http.StatusOK:
+		return json.Valid(bodyBytes), nil, nil
```

Comment:
if 200 means good key and has `read_user` scope, then why the additional `json.Valid` check?